### ISSUE TITLE
Is there any SGD learning rate schedule?

### ISSUE THREAD

**Issue by chezou** [ibayer/fastFM-core#11]
In pyFM or scikit-learn, there are some learning rate schedule strategies for SGD.
https://github.com/coreylynch/pyFM/blob/master/pyfm_fast.pyx#L281-L286
http://scikit-learn.org/stable/modules/sgd.html#id1

Are there any learning rate schedules in fastFM?

**Comment by ibayer** [comment]
Currently fastFM only contains the basic fixed step-size SGD.
More advanced SGD strategies are very interesting, especially for the BPR loss that can't be optimized with standard coordinate descent (CD).
It's just that my research focuses more on CD based solvers and I have to prioritize my time. I also don't see any benefit of SGD over CD if you don't have huge datasets. CD based solvers have the advantage that they can be easily turned into mcmc (gibbs) sampler which allows to tune (integrate out) hyper-parameter automatically.

**Comment by chezou** [comment]
Thank you for reply.
You means MCMC solver is good with less tuning parameters. I found libfm also doesn't any learning rate schedule selection.

**Comment by ibayer** [comment]
That's both correct.

**Comment by chezou** [comment]
I tried to tune SDG for ml-100k, but MCMC is superior than SGD.
